std/typecons.d

```diff
@@ -1175,7 +1175,7 @@ if (distinctFieldNames!(Specs))
             size_t h = 0;
             static foreach (i, T; Types)
             {{
-                const k = typeid(T).getHash((() @trusted => cast(const void*) &field[i])());
+                const k = hashOf((() @trusted => cast(const void*) &field[i])());
```

Comment:
This is wrong: you're hashing a pointer to the member rather than the member itself. `hashOf` takes its argument by reference (or by value in cases where passing by value is equivalent and faster).